lib/src/components/NxAccordion/stateful/__tests__/NxStatefulAccordion.test.tsx

```diff
@@ -4,46 +4,253 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import { act } from 'react-dom/test-utils';
+import React from 'react';
+import userEvent from '@testing-library/user-event';
 
-import NxStatefulAccordion, { Props } from '../NxStatefulAccordion';
+import NxStatefulAccordion from '../NxStatefulAccordion';
 import NxAccordion from '../../NxAccordion';
-import * as enzymeUtils from '../../../../__testutils__/enzymeUtils';
+import { rtlRender, rtlRenderElement } from '../../../../__testutils__/rtlUtils';
+import { render } from '@testing-library/react';
+import NxButton from '../../../NxButton/NxButton';
 
 describe('NxStatefulAccordion', function() {
-  const getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxStatefulAccordion, {});
 
-  it('renders an NxAccordion with the specified props', function() {
+  const quickRender = rtlRender(NxStatefulAccordion, {});
+  const renderEl = rtlRenderElement(NxStatefulAccordion, {});
+
+  it('renders a <details> with the specified props', function() {
     const onToggle = jest.fn(),
-        component = getShallowComponent({ onToggle, className: 'foo', id: 'bar' });
+        component = renderEl({ onToggle, id: 'bar' });
 
-    expect(component).toMatchSelector(NxAccordion);
-    expect(component).toHaveProp('onToggle', onToggle);
-    expect(component).toHaveProp('className', 'foo');
-    expect(component).toHaveProp('id', 'bar');
+    expect(component!.tagName).toBe('DETAILS');
+    expect(component).toHaveAttribute('id', 'bar');
+  });
+
+  it('sets the provided className', function() {
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' })!;
+
+    expect(customEl).toHaveClass('foo');
+
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
   it('sets the NxAccordion open prop to the defaultOpen prop initially', function() {
-    expect(getShallowComponent()).toHaveProp('open', false);
-    expect(getShallowComponent({ defaultOpen: false })).toHaveProp('open', false);
-    expect(getShallowComponent({ defaultOpen: true })).toHaveProp('open', true);
+    expect(renderEl()).not.toHaveAttribute('open');
+    expect(renderEl({ defaultOpen: false })).not.toHaveAttribute('open');
+    expect(renderEl({ defaultOpen: true })).toHaveAttribute('open');
   });
 
-  it('toggles the NxAccordion open prop when the NxAccordion onToggle callback is called', function() {
-    const component = getShallowComponent();
+  it('renders non-header children in content wrapper', function() {
```

Comment:
> was getting this console error (also seen in NxAccordion tests),

I think this is because `NxAccordion` has an unwritten rule that the `NxAccordion.Header` must be one of its immediate JSX descendants.  Your fix is good.  We should add a ticket to get that documented in the gallery page.

Reply:
Ah, that makes more sense now. Ticket created here: https://issues.sonatype.org/browse/RSC-1397